### DISCUSSION TITLE
Correct thumbnails, after clicking on them the wrong photo is displayed (for some photos)

### DISCUSSION THREAD

**Discussion by hpoperator86** [photoprism/photoprism#4200] [Q&A]
#### 1. What is not working as documented?

Hello dear community,

I currently have just over 28000 photos indexed. They are family photos and photos from everyday life. 

As the small preview image, all the photos are displayed correctly, but with some photos I have the problem that the correct photo is displayed, but when I click on it, a completely different photo is displayed. The strange thing is that there are also photos here that I have already opened once and everything was correct and when I opened them again, the same wrong photo was displayed. It seems as if the database here refers to a different photo. I am attaching a video here that shows the problem (In the source folder where the photos are located, all photos are fine).

https://github.com/photoprism/photoprism/assets/149537735/76492083-88a0-4b45-b262-8e0807594092

#### 2. How can we reproduce it?
unfortunately I have no idea, I'm not that technically savvy.
see video.

#### 3. What behavior do you expect?
the correct display of photos.

#### 4. What could be the cause of your problem?
I suspect the database

#### 5. Can you provide us with example files for testing, error logs, or screenshots?
yes, to a certain extent. These are family photos, so unfortunately I can't make all my pictures available

#### 6. Which software versions do you use?
(a) PhotoPrism Architecture & Build Number: AMD64, ARM64, ARMv7,...
PhotoPrism® CE Build 231128-f48ff16ef

(b) Database Type & Version: MariaDB, MySQL, SQLite,...
1:11.3.2+maria~ubu2204

(c) Operating System Types & Versions: Linux, Windows, Android,...
Linux 6.1.0-0.deb11.17-amd64 (PhotoPrism was installed in a Docker)

(d) Browser Types & Versions: Firefox, Chrome, Safari on iPhone,...
Firefox (latest version)

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?
Google, Translate Plugin and uBlockOrigin Plugin

#### 7. On what kind of device is PhotoPrism installed?
Fujitsu Primergy TX1310 M1 - Server

(a) Device / Processor Type: Raspberry Pi 4, Intel Core i7-3770, AMD Ryzen 7 3800X,...
Intel(R) Xeon(R) CPU E3-1226 v3 @ 3.30GHz

(b) Physical Memory & Swap Space in GB
Physical Memory: 8GB DDR3L 1600MHz ECC (HMT41GU7BFR8A-PB)
SwapSpace: 4GB

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage,...
HDD

(d) Anything else that might be helpful to know?
I have photoprism running as a docker which was installed with a docker-compose under portainer. Here in this thread I posted the docker-compose and the stack.env:
https://github.com/photoprism/photoprism/discussions/4173

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?
**No**

**Using NGINX?**
No!

**Comment by lastzero** [comment]
If there is no way to reproduce this issue, we unfortunately cannot fix it or even confirm it as a bug:
- Something like this would usually happen if changes have been performed in the background (e.g. **updated metadata such as location or time, pictures moved to the archive, ongoing index operations,...**) or there are other inconsistencies so that the pictures you are trying to open are no longer available (at least not in the latest search results, depending on which filters and sort order you are using).
- I would recommend to also test this with another browser to see if that makes a difference, because that would indicate a potential frontend issue. Thank you very much!

**Comment by lastzero** [comment]
If the problem is that the selected pictures are no longer included in the (latest) search results, the first available picture (i.e. the one at the beginning) would be displayed. To confirm, you can also press the "Reload" button to see if the displayed results have changed, for example when sorting by "Recently Added" and an indexing process is currently taking place in the background:

![Screenshot 2024-04-16 at 19 19 00](https://github.com/photoprism/photoprism/assets/301686/c435e90f-e141-420e-aca4-0e45eb1fab52)

**Comment by lastzero** [comment]
I've moved this to discussions so our community can help with troubleshooting and suggesting a solution we can implement.

**Comment by cheechm13** [comment]
Same issue for me, but only seems to be happening from a single person's phone that is uploading from Photosync on iOs. The thumbnail does not match the live photo preview.

PhotoPrism® CE
Build [231021-9abea5b55

**Comment by mtaipe** [comment]
I have the same problem as shown in the video, it is random but have seen more than 10 times. Reloading the page fixes.

No one else is removing pictures but there is an ongoing import in background, would that be the cause.

**Reply by lastzero** [reply]
An ongoing import and/or choosing a random sort order (which isn't officially supported), can definitely cause this as both can cause changes / inconsistencies in the results.

**Comment by hpoperator86** [comment]
I do not use automatic import in photoprism and have the same problem as described.
I import the photos from my smartphone into a network folder using Photosync, but I always trigger the indexing of this network folder manually from time to time. The photos in the network folder are displayed correctly when opened normally (not via Photoprism). You can also see this in the video.
The photos that have this "defect" with the wrong image in photoprism are no longer usable for photoprism, as this incorrect display is permanent.

What I have noticed, however, as already described, is that a photo that is displayed correctly in photoprism one minute ago has the same "defect" the next minute.
I suspect, as also mentioned at the beginning, that somehow the database there sets an incorrect assignment..... for whatever reason.
It is difficult for me as a lender to specify the whole thing. I only know THAT IT IS THAT WAY, but not WHY it happens. That's why I can't say how to reproduce this problem. 
I have neither updated the metadata nor moved any images, nor was any indexing running in the background. I have tested other browsers and with Chrome, Edge and Opera it is the same as with Firefox.

**Reply by lastzero** [reply]
Do you have both `PHOTOPRISM_AUTO_INDEX` and `PHOTOPRISM_AUTO_IMPORT` explicitly set to `-1` in your configuration? See:

- https://docs.photoprism.app/getting-started/config-options/#index-workers

**Reply by graciousgrey** [reply]
If you can confirm that there is no indexing/importing process running in the background while you observe this issue, can you check if reloading the page fixes the problem?

It would also be good to know in which section the problem occurs. In the search (/library/browse) or for example in the folder view (/library/folder/...)? If you observe this behavior in one section, can you reproduce it for the same photo in another section?
Have you applied any search filters? What sort order do you use when you observe the problem?

**Comment by greggking** [comment]
I have this issue and found one way it can happen: collision of the randomly generated part of the filename, e.g. `IMG_1234`.

I had a video showing the wrong thumbnail. Checking the files (select the image -> Edit -> Files tab) I found three files: the original video (`IMG_1234.heic`), a generated jpeg thumbnail that matched the video (`IMG_1234.jpeg`), and an unrelated image that shared the first part of the filename, `IMG_1234.png`. The png was being used as the thumbnail.

There is an "unstack" button in the Files tab you can use to separate the incorrect thumbnail from the other files.

**Comment by BaiMaKe** [comment]
What @greggking mentioned appears to be the driver on this. I've been running into it all across my library and its primarily affecting videos that my wife has taken with her iphone. They get tied to other images as thumbnails when the name has some matching characters. 

Is there a configuration option to tweak this?
